### PR TITLE
Make logging configurable.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2024/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2025/

File: cmd/charmd/main.go [hunk 1]
```diff
@@ -24,8 +24,7 @@ import (
 )
 
 var (
-	logger        = loggo.GetLogger("charmd")
-	loggingConfig = flag.String("logging-config", "", "specify log levels for modules e.g. <root>=TRACE")
```
Comment (mhilton) [inline cmd/charmd/main.go]: leave the flag, it's probably been set on some server somewhere. you can ignore it though.

File: config/config.go [hunk 2]
```diff
@@ -41,6 +41,7 @@ type Config struct {
 	SearchCacheMaxAge DurationString  `yaml:"search-cache-max-age,omitempty"`
 	Database          string          `yaml:"database,omitempty"`
 	AccessLog         string          `yaml:"access-log"`
+	LoggingLevel      string          `yaml:"logging-level"`
```
Comment (mhilton) [inline config/config.go:44]: Please update the test for this.

**Review by mhilton — CHANGES_REQUESTED** [review]
Generally pretty good, but I'd like a couple of changes.

**Review by jrwren — APPROVED** [review]
I'm tempted to suggest that the -logging-config option should be kept and used to override the config file option, to prevent breaking anything that uses that option, but I don't think its use is widespread enough, so nevermind. I mention it here in case I'm wrong, everyone can point and say, "shoulda... even thought about it..." and we can all cry about it together.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2026/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2027/

**Review by mhilton — APPROVED** [review]
LGTM

**Pull request closed without merging**